### PR TITLE
Crosshair in gamelab

### PR DESCRIPTION
**Pull request by islemaster**
Generalizes the App Lab crosshair and uses it in gamelab too.

I really wanted to rewrite this overlay in React while we're at it, but the v15 upgrade is taking a while so getting this done as-is, and we can do the React conversion for both later.

### CODE REVIEW

File: apps/src/templates/VisualizationOverlay.js [hunk 1]
```diff
@@ -3,8 +3,8 @@
 
 var constants = require('../constants');
 var CrosshairOverlay = require('./CrosshairOverlay');
-var gridUtils = require('./gridUtils');
-var elementUtils = require('./designElements/elementUtils');
+var gridUtils = require('../applab/gridUtils');
+var elementUtils = require('../applab/designElements/elementUtils');
```
Comment (cpirich) [inline apps/src/templates/VisualizationOverlay.js]: Does this mean that gamelab now indirectly depends on files in the applab directory? (that seems strange - and a little unexpected)

Comment (islemaster) [inline apps/src/templates/VisualizationOverlay.js]: Uhh... er... yeah.

To fix this I need to figure out a clean way to break out / inject the 'dragging' and 'element ID' behaviors for the overlay.  Or make an `AppLabVisualizationOverlay` and a `GameLabVisualizationOverlay`.  I'd prefer the former.

File: apps/src/templates/VisualizationOverlay.js [hunk 2]
```diff
@@ -64,7 +64,7 @@ VisualizationOverlay.prototype.render = function (intoElement, nextProps) {
   var oldProps = $.extend({}, this.props_);
   $.extend(this.props_, nextProps);
 
-  if (this.props_.scale !== oldProps.scale) {
+  if (this.props_.scale !== oldProps.scale || !this.screenSpaceToAppSpaceTransform_) {
```
Comment (cpirich) [inline apps/src/templates/VisualizationOverlay.js:61]: I don't follow this particular change, but perhaps it is an existing bug that you're fixing?

**Comment by cpirich** [comment]
👍 Looks good, with a few comments. Happy to see the creation of the constants file and the consistent use of constants for the width and height (especially because we may be changing those around soon!)

File: apps/src/templates/VisualizationOverlay.js (hunk 2 above)
Comment (islemaster) [inline apps/src/templates/VisualizationOverlay.js:61]: I suspect it is.  I'm not sure why this has never come up in App Lab, but when I dropped the overlay into gamelab it failed to calculate this transform on first render.

**Comment by islemaster** [comment]
@cpirich PTAL - I broke the applab-specific crosshair stuff out.  Overall diff is a bit messy but the individual commits should be easy to follow.  Let me know what you think.

**Pull request merged**